pkg/build/registry/buildconfig/rest_test.go

```diff
@@ -1,694 +0,0 @@
-	storage := REST{&mockRegistry}
-	buildConfig := mockBuildConfig()
-	_, err := storage.Create(kapi.NewDefaultContext(), buildConfig)
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
-}
-
-func mockBuildConfig() *api.BuildConfig {
-	return &api.BuildConfig{
-		ObjectMeta: kapi.ObjectMeta{
-			Name:      "data-build",
-			Namespace: kapi.NamespaceDefault,
-			Labels: map[string]string{
-				"name": "data-build",
-			},
-		},
-		Parameters: api.BuildParameters{
-			Source: api.BuildSource{
-				Type: api.BuildSourceGit,
-				Git: &api.GitBuildSource{
-					URI: "http://my.build.com/the/build/Dockerfile",
-				},
-			},
-			Strategy: api.BuildStrategy{
-				Type: api.STIBuildStrategyType,
-				STIStrategy: &api.STIBuildStrategy{
-					From: &kapi.ObjectReference{
-						Kind: "DockerImage",
-						Name: "builder/image",
-					},
-				},
-			},
-			Output: api.BuildOutput{
-				DockerImageReference: "repository/data-build",
-			},
-		},
-	}
-}
-
-func mockBuildConfigImageRef() *api.BuildConfig {
-	return &api.BuildConfig{
-		ObjectMeta: kapi.ObjectMeta{
-			Name:      "data-build",
-			Namespace: kapi.NamespaceDefault,
-			Labels: map[string]string{
-				"name": "data-build",
-			},
-		},
-		Parameters: api.BuildParameters{
-			Source: api.BuildSource{
-				Type: api.BuildSourceGit,
-				Git: &api.GitBuildSource{
-					URI: "http://my.build.com/the/build/Dockerfile",
-				},
-			},
-			Strategy: api.BuildStrategy{
-				Type: api.STIBuildStrategyType,
-				STIStrategy: &api.STIBuildStrategy{
-					From: &kapi.ObjectReference{
-						Name: "builder/image",
-					},
-				},
-			},
-			Output: api.BuildOutput{
-				DockerImageReference: "repository/data-build",
-			},
-		},
-	}
-}
-
-func TestUpdateBuildConfig(t *testing.T) {
-	mockRegistry := test.BuildConfigRegistry{}
-	storage := REST{&mockRegistry}
-	buildConfig := mockBuildConfig()
-	obj, created, err := storage.Update(kapi.NewDefaultContext(), buildConfig)
-	if err != nil || created {
-		t.Errorf("unexpected error: %v", err)
-	}
-	switch obj.(type) {
-	case *kapi.Status:
-		t.Errorf("Unexpected operation error: %v", obj)
-
-	case *api.BuildConfig:
-		if !reflect.DeepEqual(buildConfig, obj) {
-			t.Errorf("Updated build does not match input build."+
-				" Expected: %v, Got: %v", buildConfig, obj)
-		}
-	default:
-		t.Errorf("Unexpected result type: %v", obj)
-	}
-}
-
-func TestUpdateBuildConfigError(t *testing.T) {
-	mockRegistry := test.BuildConfigRegistry{Err: fmt.Errorf("Update error")}
-	storage := REST{&mockRegistry}
-	buildConfig := mockBuildConfig()
-	_, _, err := storage.Update(kapi.NewDefaultContext(), buildConfig)
-	if err != mockRegistry.Err {
-		t.Errorf("unexpected error: %v", err)
-	}
-}
-
-func TestBuildConfigRESTValidatesCreate(t *testing.T) {
-	mockRegistry := test.BuildConfigRegistry{}
-	storage := REST{&mockRegistry}
-	failureCases := map[string]struct {
-		expectSuccess bool
-		data          api.BuildConfig
-	}{
-		"blank sourceURI": {
```

Comment:
I hate to lose these tests... move them to validation_test.go in the validation package?